retrofit/src/main/java/retrofit/OkHttpCall.java

```diff
@@ -151,7 +158,8 @@ private void callSuccess(Response<T> response) {
       // If the underlying source threw an exception, propagate that rather than indicating it was
       // a runtime exception.
       catchingBody.throwIfCaught();
-      throw e;
+      throw new RuntimeException("Problem has occurred during conversion of response for "
```

Comment:
Not a fan of wrapping here. It hides the underlying error and creates RetrofitError-like semantics all over again.